Etabs_Adapter/Types/DynamicComparer.cs

```diff
@@ -0,0 +1,103 @@
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Adapters.ETABS;
+using BH.oM.Base;
+using BH.oM.Spatial.SettingOut;
+using BH.oM.Structure.Constraints;
+using BH.oM.Structure.Elements;
+using BH.oM.Structure.Loads;
+using BH.oM.Structure.MaterialFragments;
+using BH.oM.Structure.SectionProperties;
+using BH.oM.Structure.SurfaceProperties;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BH.Adapter.ETABS.Types
+{
+        public class DynamicComparer : IEqualityComparer<IBHoMObject>
+        {
+
+            // Use of STREAMS and REFLECTIONS
+
+            // 1. Equality based on ETABS obj Id or Name
+            public bool Equals(IBHoMObject obj1, IBHoMObject obj2)
+            {
+                if (obj1 == null || obj2 == null) return false;
+                if (obj1.GetType() != obj2.GetType()) return false;
+
+
+                Type objType = obj1.GetType();
+
+                // For physical objects, use Id as it is never null
+                if (objType == typeof(Node) || objType == typeof(Bar) || objType == typeof(Panel) || objType == typeof(RigidLink) || objType == typeof(FEMesh))
+                    return GetEtabsId(obj1).Id.ToString() == GetEtabsId(obj2).Id.ToString();
+                // For all other items, use Name as it is never null
+                if (objType == typeof(ISectionProperty) || objType == typeof(IMaterialFragment) || objType == typeof(ISurfaceProperty) ||
+                    objType == typeof(Loadcase) || objType == typeof(LoadCombination) || objType == typeof(ILoad) || objType == typeof(LinkConstraint) ||
+                    objType == typeof(Level) || objType == typeof(Grid))
+                    return obj1.Name.ToString() == obj2.Name.ToString();
```

Comment:
We already have all these comparers defined in the Adapter - can we not make sure of that rather than rewriting it here:

https://github.com/BHoM/ETABS_Toolkit/blob/edc11e02ca0a6e0b06a6a9a721e7c98a9a380295/Etabs_Adapter/Types/Comparer.cs#L57-L62

Reply:
@peterjamesnugent, thanks for the comment.
It would be great to reuse it yeah.
Only thing is that there are some objects that are not dealt with by the Adapter Comparer (e.g. Level, FEMesh,LoadCase...) - see screenshot below.
Can we add them in it?

![image](https://github.com/user-attachments/assets/a6aa5d4b-847d-42da-9edd-701b7a8f5a39)

Reply:
Yes, I don't think that's an issue adding them to the `Comparer.cs` - most of them will be `NameOrDescriptionComparer()` anyway.

Can you add to the test script, trying to push two objects with the same name for each type you add (just to test that we are not breaking anything).